### ISSUE TITLE
Updating via kubectl -f apply does not handle options properly

### ISSUE THREAD

**Issue by pavolloffay** [jaegertracing/jaeger-operator#304]
Changing options (storage, query, collector..) in yaml CR and applying change via `kubectl applye` does not work as expected. The updated options get duplicated:

```yaml
    options:
      es:
        bulk:
          flush-interval: 200ms
          size: 10
          workers: 1
      es.bulk.flush-interval: 200ms
      es.bulk.size: "5"
      es.bulk.workers: "1"
```

Created CR - first with `es.bulk.size=5` then changed to 10
```yaml
apiVersion: jaegertracing.io/v1
kind: Jaeger
metadata:
  name: simple-prod
spec:
  strategy: production
  storage:
    type: elasticsearch
    options:
      es:
        bulk:
          size: 10
          workers: 1
          flush-interval: 200ms
    elasticsearch:
      nodeCount: 1
      resources:
    esIndexCleaner:
      enabled: true
      schedule: "*/30 * * * *"
      numberOfDays: 1
```

**Comment by pavolloffay** [comment]
I think we will have to use `FreeForm` otherwise we don't know how to merge these properties.

**Comment by objectiser** [comment]
That might have an affect on users already using the condensed form of a property. But if the only solution, then needs to be listed as breaking change for v1.

**Comment by pavolloffay** [comment]
I don't think it would break users using condensed form or properties. Json `{"a": tue, "a.b":true}` can be handled without problems unless I miss something. 

I think we have two options change Options to FeeForm (`[]byte`) or change it to `map[string]string`. The latter is easier to work with if we need to change properties in the operator.

**Comment by objectiser** [comment]
Fixed by #523
